### ISSUE TITLE
Better matching?

### ISSUE THREAD

**Issue by rakslice** [sammertens/fujitsu_scroll#2]
I have a Fujitsu Lifebook P772/G that has the scroll wheel.

This machine's DMI information has as a product name is only its specific SKU `FMVNP8AE`, rather than `LIFEBOOK P772` or similar.

Just adding a matcher for `FMVNP8AE` works, i.e.

```
	{                                                            
                .matches = {                                         
                        DMI_MATCH(DMI_SYS_VENDOR, "FUJITSU"),        
                        DMI_MATCH(DMI_PRODUCT_NAME, "FMVNP8AE"),     
                },                                                   
        }, 
```
But it's tedious to have a rule like this for every SKU of laptop with the scroll wheel if they can even be found; Is there a better way to match that is still sufficiently selective, like matching every Fujitsu that is a laptop and is of a certain era somehow?

**Comment by sammertens** [comment]
I have no idea how widespread this device is/was.  I know Fujitsu tried several different hardware implementations of their scroll wheel before (apparently) giving up entirely.  They were trying to sell a reference kit for a while, maybe to entice other manufacturers, but I don’t think they had much success.  I know that it talks to the i8042, so safe to say nobody’s putting it in anything new.At first, I only knew it existed in the T901 I had picked up on EBay.  Asking Fujitsu about it directly gave me nothing.  I only know about the 772 because somebody said it works on theirs. Your model makes 3.Yeah, a big fat whitelist is going to be increasingly unwieldy.  Two checks are actually done - first, the DMI check.  Then, the driver checks the response to a probe sequence.  That probe sequence happens to be the same as the one used for the Synaptics touchpad on my T901, and that’s how I was able to get in and start reverse engineering the protocol; the scroll wheel simply responds with different data than the touchpad does.  In the kernel code there are notes about how some hardware can get messed up by it, so the Synaptics probe is only done after other hardware is checked for.  I guess that means it ought to be “safe enough” to probe for the Fujitsu scroll wheel any time it’s safe enough to probe for a Synaptics touchpad.  I was thinking about making the DMI check optional with a kernel configuration option anyway.I’ll see about trying to carve out some time this weekend for this and the sensitivity issue.- SamOn May 26, 2023, at 7:21 PM, rakslice ***@***.***> wrote:﻿
I have a Fujitsu Lifebook P772/G that has the scroll wheel.
This machine's DMI information has as a product name is only its specific sku FMVNP8AE, rather than LIFEBOOK P772 or similar.
Just adding a matcher for FMVNP8AE works, i.e.
	{                                                            
                .matches = {                                         
                        DMI_MATCH(DMI_SYS_VENDOR, "FUJITSU"),        
                        DMI_MATCH(DMI_PRODUCT_NAME, "FMVNP8AE"),     
                },                                                   
        }, 

But it's tedious to have a rule like this for every SKU with the scroll wheel if they can even be found; Is there a better way to match that is still sufficiently selective, like matching every Fujitsu that is a laptop and is of a certain era somehow?
